flambe/learn/train.py

```diff
@@ -229,14 +319,21 @@ def _train_step(self) -> None:
                 # Update iter scheduler
                 if self.iter_scheduler is not None:
                     learning_rate = self.iter_scheduler.get_lr()[0]  # type: ignore
-                    log(f'{tb_prefix}Training/LR', learning_rate, global_step)
+                    log(f'{log_prefix}/LR', learning_rate, global_step)
                     self.iter_scheduler.step()  # type: ignore
 
-            # Zero the gradients when exiting a train step
+                # Zero the gradients when exiting a train step
                 self.optimizer.zero_grad()
+                # logging train metrics
+                if self.extra_training_metrics_log_interval > self._last_train_log_step:
+                    self._log_metrics(log_prefix, metrics_with_states, global_step)
+                    self._last_train_log_step = i
+            if self._last_train_log_step != i:
+                # log again at end of epoch, if not logged at the end of epoch before
```

Comment:
is what you are calling an epoch here actually an epoch? Looks like it's at the end of a step rather, which can be any arbitrary number of iterations, as set by ``iter_per_step`` or ``epoch_per_step``